bazel/foreign_cc/BUILD

```diff
@@ -13,12 +42,46 @@ cmake_external(
         "CMAKE_BUILD_TYPE": "RelWithDebInfo",
     },
     cmake_options = ["-GNinja"],
+    generate_crosstool_file = select({
+        ":windows_x86_64": True,
+        "//conditions:default": False,
+    }),
     lib_source = "@com_github_c_ares_c_ares//:all",
     make_commands = [
         "ninja",
         "ninja install",
     ],
-    static_libraries = ["libcares.a"],
+    postfix_script = select({
+        ":windows_dbg_build": "cp $BUILD_TMPDIR/CMakeFiles/c-ares.dir/c-ares.pdb $INSTALLDIR/lib/c-ares.pdb",
+        "//conditions:default": "",
+    }),
+    static_libraries = select({
+        ":windows_x86_64": ["cares.lib"],
+        "//conditions:default": ["libcares.a"],
+    }),
+)
+
+cmake_external(
```

Comment:
Can we create an `envoy_cmake_external` macro that hides the boiler plate?

Reply:
The real boiler plate part is `generate_crosstool_file`, why this is needed @sesmith177? If it is a general issue probably upstream to `rules_foreign_cc`?

Reply:
According to the README (and our testing), you need to explicitly specify the option to generate the CMake crosstool file for `rules_foreign_cc` to work on Windows

Reply:
The boilerplate to me is both the `postfix_script` library retrieval of PDB and the crosstool code. Is there a better way to support them in `rules_foreign_cc`? CC @irengrig.

We also have the ninja configuration as boilerplate. Hence why I reckon it would be useful to make a `envoy_cmake_external` and hide some of this away, but I agree with @lizan we could push further back into `rules_foreign_cc` some more standard things.